<!--REQUEST_ID=b1329ef6-89b0-4595-bb52-08b3023b02d7-->
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.48314&attempt_id=fce4ce12-79ff-470a-abdd-7aab52503dbe)

Attempted: sonic-pi

<details><summary>Partial log (click to expand)</summary><p>

```
g++: error: unrecognized command line option '-msse'; did you mean '-fdse'?
make[2]: *** [external_libraries/CMakeFiles/boost_program_options.dir/build.make:180: external_libraries/CMakeFiles/boost_program_options.dir/boost/libs/program_options/src/variables_map.cpp.o] Error 1
g++: error: unrecognized command line option '-mfpmath=sse'
g++: error: unrecognized command line option '-msse2'
make[2]: *** [external_libraries/CMakeFiles/boost_program_options.dir/build.make:193: external_libraries/CMakeFiles/boost_program_options.dir/boost/libs/program_options/src/winmain.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1058: external_libraries/CMakeFiles/boost_program_options.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
builder for '/nix/store/vqgydq67da9wvwy7hnblv6crlv8478y8-supercollider-3.9.3.drv' failed with exit code 2
cannot build derivation '/nix/store/77fj6p7sdmz11b1qm78dy410sm1p0pg5-sonic-pi-3.1.0.drv': 1 dependencies couldn't be built
error: build of '/nix/store/77fj6p7sdmz11b1qm78dy410sm1p0pg5-sonic-pi-3.1.0.drv' failed
```
</p></details>

--- a/pkgs/applications/audio/sonic-pi/default.nix
+++ b/pkgs/applications/audio/sonic-pi/default.nix
@@ -20,14 +20,14 @@ let
   };
 
 in stdenv.mkDerivation rec {
-  version = "3.0.1";
+  version = "3.1.0";
   name = "sonic-pi-${version}";
 
   src = fetchFromGitHub {
     owner = "samaaron";
     repo = "sonic-pi";
     rev = "v${version}";
-    sha256 = "1l1892hijp1dj2h799sfjr699q6xp660n0siibab5kv238521a81";
+    sha256 = "0gi4a73szaa8iz5q1gxgpsnyvhhghcfqm6bfwwxbix4m5csbfgh9";
   };
 
   buildInputs = [
@@ -58,21 +58,21 @@ in stdenv.mkDerivation rec {
     export SONIC_PI_HOME=$TMPDIR
     export AUBIO_LIB=${aubio}/lib/libaubio.so
 
-    pushd app/server/bin
+    pushd app/server/ruby/bin
       ./compile-extensions.rb
       ./i18n-tool.rb -t
     popd
 
     pushd app/gui/qt
       cp -f ruby_help.tmpl ruby_help.h
-      ../../server/bin/qt-doc.rb -o ruby_help.h
+      ../../server/ruby/bin/qt-doc.rb -o ruby_help.h
 
       substituteInPlace SonicPi.pro \
         --replace "LIBS += -lrt -lqt5scintilla2" \
                   "LIBS += -lrt -lqscintilla2 -lqwt"
 
       lrelease SonicPi.pro
-      qmake SonicPi.pro 
+      qmake SonicPi.pro
 
       make
     popd

--- a/pkgs/development/interpreters/supercollider/default.nix
+++ b/pkgs/development/interpreters/supercollider/default.nix
@@ -34,6 +34,6 @@ stdenv.mkDerivation rec {
     description = "Programming language for real time audio synthesis";
     homepage = http://supercollider.sourceforge.net/;
     license = stdenv.lib.licenses.gpl3Plus;
-    platforms = stdenv.lib.platforms.linux;
+    platforms = [ "x686-linux" "x86_64-linux" ];
   };
 }

